### ISSUE TITLE
Crashes when entering fullscreen from a shortcut window on macOS

### ISSUE THREAD

**Issue by dvdhoo** [brave/brave-browser#16827]
## Description 
After creating a shortcut app window on macOS, whne clicking on the fullscreen button, Brave crashes immediately.

## Steps to Reproduce

   1. create a shortcut window (select open in window option), or open an existing shortcut in window
   2. click fullscreen button on the top left
   3. Brave crashes

## Actual result:
<img width="827" alt="截屏2021-07-08 00 50 01" src="https://user-images.githubusercontent.com/691461/124864171-77981780-df86-11eb-8dd7-f2094c037513.png">

clicking on the green fullscreen button and it crashes Brave

## Expected result:
app window opens in fullscreen

## Reproduces how often: 
100% of the time

## Brave version (info found on brave://version)
V1.26.74

## Miscellaneous Information:
macOS Big Sur 11.4
MacBook Pro 13-inch 2020 Intel (4 Thunderbolt 3 ports)

**Comment by bsclifton** [comment]
@dvdhoo I tried to reproduce this and didn't have any luck. I'm on Big Sur on an M1 Mac Mini and I did the following steps:
1. Had version 1.26.74
2. Visited Google (since you showed this) and did a search
3. Right click any result and pick `Open in a new window`
4. Hit green button on the new window

Can you confirm these are the steps you're trying?

**Comment by dvdhoo** [comment]
@bsclifton No. You will need to create a shortcut. Follow  "Menu -> More tools -> Create Shortcut...", and make sure you select "Open in Window". Then you can try click on fullscreen. Not sure if this is going to happen on M1 mac though.

**Comment by iefremov** [comment]
@dvdhoo could you please check in latest brave? This could've been fixed by migrating to chromium 92

**Comment by kltb** [comment]
I get this crash still if Yabai is running.

**Comment by iefremov** [comment]
@kltb gotcha, thanks. What's on your brave://version? And could you please check what you have on brave://crashes?

**Comment by kltb** [comment]
Rereading this issue I have something separate that I should create as a new issue which is why I deleted my original comment but I'll put my message regardless. 

@iefremov  I've reinstalled Yabai to check this.

Steps to reproduce behaviour:
- Have Yabai installed
- On a webpage, open hamburger menu > More Tools > Create Shortcut > Tick Open in new window. 
- Any attempt to open the shortcut will cause a spinning beach ball

This gives me a spinning beach ball and MacOS reporting Brave as not responding but doesn't 'crash' Brave usually. I have a crash report from yesterday that might be helpful though: `9f902300-217a-ae06-0000-000000000000`

Brave | 1.27.109 Chromium: 92.0.4515.115 (Official Build) (arm64)
Revision | 48cb2f4029b84b003719740a6cf9ca73f374a857-refs/branch-heads/4515_105@{#4}
OS | macOS Version 11.4 (Build 20F71)
